### ISSUE TITLE
No background on Cover block creates a11y issue

### ISSUE THREAD

**Issue by troychaplin** [WordPress/gutenberg#65868]
### Description

When adding a cover block and using light coloured text and a low opacity, the lack of background colour on the primary block wrapper leads to an a11y error. Currently the opacity is what appears influence a pass of a fail.

## Suggested solution:

- add background color option to cover block and set it to #111111 by default when block is placed on page.

### Step-by-step reproduction instructions

I have two ways to easily create a contrast error, both tested in 6.7 and the latest trunk of TT5 as of an hour ago. Both approaches were tested with [HTML_CodeSniffer](https://squizlabs.github.io/HTML_CodeSniffer/) and Firefox Dev Edition and the [Wave Accessibility Extension](https://addons.mozilla.org/en-CA/firefox/addon/wave-accessibility-tool/).

#### Approach 1:

- add cover block
- set opacity to 60

#### Approach 2:

- add cover block
- use dark image
- remove opacity
- use white (or bright yellow) text

In both cases using the inspector adding a background of black to the main cover div fixes the issue.

### Screenshots, screen recording, code snippet

![Image](https://github.com/user-attachments/assets/2d9cd85c-3637-4e62-979a-06c417eb9b93)
![Image](https://github.com/user-attachments/assets/c2e483ed-9871-4bca-bb70-9229f03a9aeb)
![Image](https://github.com/user-attachments/assets/97b53bfd-811d-4a1b-9c94-bf8b30e74ea1)
![Image](https://github.com/user-attachments/assets/00089faa-7ef5-4fc4-9190-262d075b379c)

### Environment info

- WP 6.7, Gutenberg 19.3.0, TwentyTwentyFive lastest trunk as of Oct 3rd @ 10pm EST
- Impacts all browsers
- Is not system specific

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by carolinan** [comment]
I believe this is a duplicate of https://github.com/WordPress/gutenberg/issues/39692?

**Comment by richtabor** [comment]
It's worth noting that this is dependent on the theme's background color (or the background color of the element the cover block is placed in. It's not possible to account for all variations and setting a fixed background color for the default state of cover block would not be an appropriate solution. 

For resetting, there's a bit more of a path. **Perhaps when resetting the image (removing it), the overlay value is set higher to engage the light/dark contrast, say `70`.** 


![Image](https://github.com/user-attachments/assets/c1dd84eb-c3a4-40e4-b468-c6e4fd58f0da)
![Image](https://github.com/user-attachments/assets/9c2938d6-d2a1-4ea7-b516-37a52e0809fe)

**Comment by troychaplin** [comment]
> It's worth noting that this is dependent on the theme's background color (or the background color of the element the cover block is placed in. It's not possible to account for all variations and setting a fixed background color for the default state of cover block would not be an appropriate solution.

I'm not sure there is another solution @richtabor. If you drop a cover block on a page, select a dark color and use white text, this will still fail a11y. The overlay is not taken into consideration here, the parent wrapper is what is checked, and in this case the cover block does not have the ability to do that without wrapping it in a group block, or other some other block.

Here's some examples of 3 configs that won't pass a11y

<img width="2138" height="1244" alt="Image" src="https://github.com/user-attachments/assets/cd55a402-b9be-4cef-8bc3-fccca253b471" />
<img width="2106" height="1178" alt="Image" src="https://github.com/user-attachments/assets/6f0a4859-f565-4042-86e1-7d28ef2fca95" />
<img width="2048" height="994" alt="Image" src="https://github.com/user-attachments/assets/73c1f5bf-47fd-449d-aef6-066e3075a435" />

To your point, when used inside another block, such as the group block, and it's bg is dark, there is no issue. But I wouldn't expect users to think or even know to do this.

<img width="2116" height="1308" alt="Image" src="https://github.com/user-attachments/assets/3c9e8a1d-fd92-423f-b2c9-4acc85cce860" />

Also to your point, the theme's bg will impact this as well

<img width="2118" height="1298" alt="Image" src="https://github.com/user-attachments/assets/85a7f0a6-30a0-4999-b977-0fe2dda0df98" />
<img width="2128" height="1280" alt="Image" src="https://github.com/user-attachments/assets/ab587214-16a3-436b-a133-430634c1cfcf" />

I also find the contrast warning misleading and potentially confusing to users. From their point of view, it's white text on a dark image, where's the issue.

<img width="2024" height="958" alt="Image" src="https://github.com/user-attachments/assets/e8b47c76-4bda-4fff-91a3-13c1995b51fd" />

I'm not sure what the ultimate solution is here, as this doesn't factor in that the user can modify the colour of the text on the cover block, or individually in innerblocks, and easily introduce other issues. But I do think a good start would be considering a background option on the cover block.
